src/Components/Components/src/Rendering/KeyedItemInfo.cs

```diff
@@ -0,0 +1,13 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace Microsoft.AspNetCore.Components.Rendering
+{
+    // Used internally during diffing to track what we know about
+    // keyed items and their positions
+    internal struct KeyedItemInfo
```

Comment:
Does this need to be mutable? Immutable is nice where possible since it's less error prone

Reply:
Previously I was OK with the mutability because this type is only used in a very localised way within `RenderTreeDiffBuilder` and doesn't really get passed around from one place to another.

However on further consideration, it's not really a big chore to make it immutable, and the perf cost of a bit more copying isn't going to hurt us (it only happens in the specific case where we detect a moved frame).

So, done.